### PR TITLE
refactor(patterns/form): update documentation of checkbox, datepicker and text-input

### LINKED ISSUE #708: No label for date picker input
# 🐞 Pattern bug report

### Where did you experience this bug ?
	
- [ ] On sketch file
- [x] On Documentation
- [x] On source code

### Description

There is no `<label>` for date picker input.

### PR DESCRIPTION
**Pull request by tiloyi**
Fix #708

## I have read [the contributing guidelines](https://mozaic.adeo.cloud/Contributing/)

- [x] Yes
- [ ] No

## Does this PR introduce a [breaking change](https://mozaic.adeo.cloud/Contributing/Developers/GitConventions/#breaking-changes-)?

- [ ] Yes
- [x] No

## Describe the changes

**Update the documentation of the Datepicker component**

GitHub issue number or Jira issue URL: [issue-708](https://github.com/adeo/mozaic-design-system/issues/708)

### CODE REVIEW

File: src/docs/Components/Form/Datepicker/code.mdx [hunk 1]
```diff
@@ -5,31 +5,47 @@ order: 2
...
+    id="example"
+    name="example"
+  />
+</div>
 ```
 
 <Highlight theme="warning" title="date picker appearance">
 
-  Using the `type="date"` attribute on an `input` element triggers the automatic generation of a date picker by the browser. The appearance of this date picker will therefore depend on the browser used.
+  Using the `type="date"` attribute on an `input` element triggers the automatic generation of a date picker by the browser. The appearance and overall functionality of this date picker will therefore depend on the browser used.
```
Comment (mohamedMok) [inline src/docs/Components/Form/Datepicker/code.mdx:46]: `and overall features`

File: src/docs/Components/Form/Datepicker/code.mdx [hunk 2]
```diff
@@ -5,31 +5,47 @@ order: 2
...
 
-  To solve this problem, you will have to implement a JavaScript library of your choice (see below "Custom date picker").
+  So, if you want to have advanced control over appearance and functionality, you will have to use a JavaScript library for that (see below "[Custom usage](/Components/Form/Datepicker/code/#custom-usage)").
```
Comment (mohamedMok) [inline src/docs/Components/Form/Datepicker/code.mdx:48]: => `To have advanced control over appearance and features...`

File: src/docs/Components/Form/Datepicker/code.mdx [hunk 3]
```diff
@@ -45,14 +61,26 @@ In addition, the `input` element must also have the `mc-text-input` class.
 
 <Preview path="default" />
 
+### Variations
+
+You can use any of the existing text-input component variations.
```
Comment (mohamedMok) [inline src/docs/Components/Form/Datepicker/code.mdx:66]: `You may use all existing text-input component variations`

File: src/docs/Components/Form/Datepicker/code.mdx [hunk 4]
```diff
@@ -45,14 +61,26 @@ In addition, the `input` element must also have the `mc-text-input` class.
...
+### Behaviors
+
+The states applicable to the date picker are identical to those applicable to the text-input component.
+
+Read the [Text input component documentation](/Components/Form/TextInput/code/#behaviors) for more information.
+
 ## Custom usage
 
-The use of the `type="date"` attribute creates a concern for consistency in the appearance of the date picker, as explained above. The way to solve this problem is to :
+The use of the `type="date"` attribute creates a concern for consistency in the appearance and functionality of the date picker, as explained above. The way to solve this problem is to:
```
Comment (mohamedMok) [inline src/docs/Components/Form/Datepicker/code.mdx:78]: `and feature`

File: src/docs/Components/Form/Field/accessibility.mdx [hunk 5]
```diff
@@ -0,0 +1,125 @@
+---
+title: 'Accessibility'
+order: 3
+---
+
+## Accessibility and semantic
+
+### Always use field inside a form
```
Comment (mohamedMok) [inline src/docs/Components/Form/Field/accessibility.mdx:8]: `You must use a field inside a form`

File: src/docs/Components/Form/TextInput/accessibility.mdx [hunk 6]
```diff
@@ -0,0 +1,69 @@
+---
+title: 'Accessibility'
+order: 3
+---
+
+## Accessibility and semantic
+
+### Always use a label with an input.
```
Comment (mohamedMok) [inline src/docs/Components/Form/TextInput/accessibility.mdx:8]: `You must use a label with an input`

**Review by mohamedMok — APPROVED** [review]
Looking good 💯
small comments about the doc

**Review by nherwegh — APPROVED** [review]

File: src/docs/Components/Form/Datepicker/code.mdx (hunk 1 above)
Comment (tiloyi) [reply src/docs/Components/Form/Datepicker/code.mdx:46]: fixed

File: src/docs/Components/Form/Datepicker/code.mdx (hunk 2 above)
Comment (tiloyi) [reply src/docs/Components/Form/Datepicker/code.mdx:48]: fixed

File: src/docs/Components/Form/Datepicker/code.mdx (hunk 3 above)
Comment (tiloyi) [reply src/docs/Components/Form/Datepicker/code.mdx:66]: fixed

File: src/docs/Components/Form/Datepicker/code.mdx (hunk 4 above)
Comment (tiloyi) [reply src/docs/Components/Form/Datepicker/code.mdx:78]: fixed

**Pull request merged**